src/lib.rs

```diff
@@ -12,16 +12,48 @@ use bdk::database::{AnyDatabaseConfig, ConfigurableDatabase};
 use bdk::keys::bip39::{Language, Mnemonic, WordCount};
 use bdk::keys::{DerivableKey, ExtendedKey, GeneratableKey, GeneratedKey};
 use bdk::miniscript::BareCtx;
-use bdk::wallet::AddressIndex;
 use bdk::{BlockTime, Error, FeeRate, SignOptions, Wallet as BdkWallet};
-use std::convert::TryFrom;
+use std::convert::{TryFrom, From};
 use std::str::FromStr;
 use std::sync::{Arc, Mutex, MutexGuard};
 
 uniffi_macros::include_scaffolding!("bdk");
 
 type BdkError = Error;
 
+// we redefine a simpler version of bdk::wallet::AddressInfo
+// it has an `address` field of type String instead of the `address` field of type `Address`
+// defined in bitcoin::util::address::Address
+pub struct AddressInfo {
+    pub index: u32,
+    pub address: String,
+}
+
+impl From<bdk::wallet::AddressInfo> for AddressInfo {
+    fn from(x: bdk::wallet::AddressInfo) -> AddressInfo {
+        AddressInfo {
+            index: x.index,
+            address: x.address.to_string()
+        }
+    }
+}
+
+// we redefine a simpler version of bdk::wallet::AddressIndex
+// only keeping the `New` and `LastUnused` variants of the enum
+pub enum AddressIndex {
+    New,
+    LastUnused,
+}
+
+impl From<AddressIndex> for bdk::wallet::AddressIndex {
+    fn from(x: AddressIndex) -> bdk::wallet::AddressIndex {
+        match x {
+            AddressIndex::New => bdk::wallet::AddressIndex::New,
+            AddressIndex::LastUnused => bdk::wallet::AddressIndex::LastUnused
```

Comment:
nitpick: we can also alias `bdk::wallet::AddressIndex` as `BdkAddressIndex`

Reply:
I like it for consistency with the others!